policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -42,7 +42,14 @@ struct NamespaceIndex {
 
 #[derive(Debug)]
 struct Namespace {
-    service_routes: HashMap<ServicePort, ServiceRoutes>,
+    // A map of Service name and port to all HttpRoutes which target that
+    // Service and port specifically.
+    service_port_routes: HashMap<ServicePort, ServiceRoutes>,
```

Comment:
This comment isn't really the whole story... we don't just hold the HttpRoutes. We also hold the channels so we can publish state to clients.

Specifically, if a client requests routes for a service:port and we don't have any known service or port for it, we'll still create an entry in this map for that service and port so that, if it is created later, we are able to publish updates to it.